coord/src/module.c

```diff
@@ -1247,11 +1246,12 @@ static int searchResultReducer(struct MRCtx *mc, int count, MRReply **replies) {
 
 cleanup:
   if (rCtx.pq) {
-    searchResult *sr;
-    while ((sr = heap_poll(rCtx.pq))) {
-      rm_free(sr);
-    }
-    heap_free(rCtx.pq);
+    heap_destroy(rCtx.pq);
+  }
+  if (rCtx.reduceSpecialCaseCtx &&
+      rCtx.reduceSpecialCaseCtx->specialCaseType == SPECIAL_CASE_KNN &&
+      rCtx.reduceSpecialCaseCtx->knn.pq) {
+    heap_destroy(rCtx.reduceSpecialCaseCtx->knn.pq);
```

Comment:
```suggestion
  if (rCtx.reduceSpecialCaseCtx->knn.pq) {
    heap_destroy(rCtx.reduceSpecialCaseCtx->knn.pq);
```

Reply:
Why can I skip checking these conditions?

Reply:
WRT adding a test: I'm not sure how to reproduce this. It was a flake timeout test that depended on the stage at which the timeout occurred, the leak could or could not happen.

Reply:
@GuyAv46 is there a case where `rCtx.reduceSpecialCaseCtx->knn.pq` exists and should not be freed?

Reply:
No, but what if `rCtx.reduceSpecialCaseCtx` was not initialized? Or is initialized not on special case KNN (the heap is a part of a union)